packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx

```diff
@@ -32,14 +37,70 @@ export const getMapFeatureInfoToolItemDef = (): ToolItemDef =>
       await IModelApp.tools.run(MapFeatureInfoTool.toolId);
     },
     badgeType: BadgeType.TechnicalPreview,
+    isHidden: new ConditionalBooleanValue(() => {
+      // Hide the MapFeatureInfoTool if the Map Layers toggle is off or no ArcGISFeature layers are active
+      let isHidden = true;
+      if (!FeatureInfoUiItemsProvider.mapLayerProviders) {
+        FeatureInfoUiItemsProvider.loadMapLayerProviders();
+      }
+      FeatureInfoUiItemsProvider.mapLayerProviders?.forEach((provider) => {
+        if (provider?.supportsMapFeatureInfo) {
+          isHidden = false;
+          return;
```

Comment:
Even though there is return here, `forEach ` will continue to iterate on the remaining imagery providers, which I think is useless.